analytic_tag_dimension/models/account_analytic_tag.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2017 PESOL (http://pesol.es) - Angel Moya (angel.moya@pesol.es)
+# Copyright 2020 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, api, fields, _
+from odoo.exceptions import ValidationError
+
+
+class AccountAnalyticTag(models.Model):
+    _inherit = 'account.analytic.tag'
+
+    analytic_dimension_id = fields.Many2one(
+        comodel_name='account.analytic.dimension',
+        string='Dimension')
+
+    @api.multi
+    def get_dimension_values(self):
+        values = {}
+        for tag in self.filtered('analytic_dimension_id'):
+            values.update({
+                tag.analytic_dimension_id.get_field_name(): tag.id,
+            })
+        return values
+
+    def _check_analytic_dimension(self):
+        tags_with_dimension = self.filtered('analytic_dimension_id')
+        dimensions = tags_with_dimension.mapped('analytic_dimension_id')
+        if len(tags_with_dimension) != len(dimensions):
+            raise ValidationError(
+                _("You can not set two tags from same dimension."))
+
+    def write(self, vals):
+        if 'analytic_dimension_id' in vals:
+            Dimension = self.env['account.analytic.dimension']
+            _models = [self.env[m] for m in Dimension.get_model_names()]
+            for tag in self.filtered('analytic_dimension_id'):
+                old_field = tag.analytic_dimension_id.get_field_name()
+                new_field = Dimension.browse(
+                    vals['analytic_dimension_id']).get_field_name()
+                if old_field == new_field:  # pragma: no cover
+                    continue
+                # Filter to avoid update report models
```

Comment:
```suggestion
            for tag in self.filtered('analytic_dimension_id'):
                Dimension = self.env['account.analytic.dimension']
                old_field = tag.analytic_dimension_id.get_field_name()
                new_field = Dimension.browse(
                    vals['analytic_dimension_id']).get_field_name()
                if old_field == new_field:  # pragma: no cover
                    continue
                # Filter to avoid update report models
                 _models = [self.env[m] for m in Dimension.get_model_names()]
```

Reply:
I don't get why...

Reply:
Only small details to improve code because `Dimension = self.env['account.analytic.dimension']` not need to load before filtered, and `_models = [self.env[m] for m in Dimension.get_model_names()]` similar.

Reply:
It's done for performance. Changing it outside the loop, you avoid to call it N times.